lib/pbio/drv/imu/imu_lsm6ds3tr_c_stm32.c

```diff
@@ -213,6 +213,9 @@ static PT_THREAD(pbdrv_imu_lsm6ds3tr_c_stm32_init(struct pt *pt)) {
         .int1_drdy_g = 1,
     }));
 
+    // Enable rounding mode so we can get gyro + accel in one read.
+    PT_SPAWN(pt, &child, lsm6ds3tr_c_rounding_mode_set(&child, ctx, LSM6DS3TR_C_ROUND_GY_XL));
```

Comment:
This is not needed. We only read 12 bytes so we never hit the rounding point.

Reply:
Alternately to actually take advantage of this, we would need to use different I2C functions to write the register once outside of the loop and then read inside of the loop (`HAL_I2C_Master_Transmit_IT` and `HAL_I2C_Master_Receive_IT` or possibly `HAL_I2C_Master_Sequential_Receive_IT`?) and a goto after error recovery to write the register again.

Reply:
Or something like the note above with no loop at all?

Reply:
Doing more in the interrupt handlers sounds dangerous to me. For example, if we call HAL functions from there, they may have (short) blocking busy loop waits that we don't know about. The less we do in interrupt handlers the better.

And I don't see how that would make using the rounding function work either.